libraries/AP_Motors/AP_MotorsHeli_RSC.cpp

```diff
@@ -190,3 +184,37 @@ void AP_MotorsHeli_RSC::write_rsc(float servo_out)
         SRV_Channels::set_output_scaled(_aux_fn, (uint16_t) (servo_out * 1000));
     }
 }
+
+    // calculate_desired_throttle - uses throttle curve and collective input to determine throttle setting
+float AP_MotorsHeli_RSC::calculate_desired_throttle(float collective_in, float thrcrv[5])
+{
+    float thrcrv_poly[5][4];
+    float throttle = 0.0f;
+    float inpt, a, b;
+
+    splinterp5(thrcrv,thrcrv_poly);
+
+    inpt = collective_in * 4.0f + 1.0f;
```

Comment:
this set of if() statements could be done more neatly like this:

uint8_t idx = constrain_int16(int8_t(collective_in * 4), 0, 3);
a = inpt - (idx+1);
b = 1.0 - inpt + 1;
throttle = thrcrv_poly[idx][0] * a + thrcrv_poly[idx][1] * b + thrcrv_poly[idx][2] * (powf(a,3.0f) - a) / 6.0f + thrcrv_poly[idx][3] * (powf(b,3.0f) - b) / 6.0f;

also note that I used powf() not pow(). pow() is double precision, and very slow